### DISCUSSION TITLE
Remove serial support from Pi4J V2?

### DISCUSSION THREAD

**Discussion by FDelporte** [Pi4J/pi4j#308] [Polls]
When experimenting with a serial device to control LED strips (see https://pi4j.com/examples/jbang/pixelblaze_output_expander/), I found out Pi4J doesn't support the required baud rate. Instead, I used the jSerialComm library. You can find more info about it at https://fazecast.github.io/jSerialComm/.

@eitch proposed removing the serial implementation from Pi4J V2 as jSerialComm is actively supported and provides excellent implementations. Focusing on the GPIO functionality for RPi has been the primary goal of Pi4J V2, and indeed, serial communication could be considered as not being part of that primary goal.

**Comment by taartspi** [comment]
Have not used the Uart myself, but post #272 caused me to do some reading to respond. Terse detail: Pigpio does not support all the function available in the SOC hardware.    I think deprecate at the first convenience then removal at the next release.  Don't know who is really using it at present but I think some time marked as deprecated is needed.  The web page can then point to articles/project Frank and Robert created as the suggested replacement.                   I know the selection I pushed says deprecate next release which is the norm, but not sure enough people will care if we announce this sooner.

**Reply by mirage22** [reply]
We use this feature in the Robo4j project for multiple hardware units. We have currently migrated Robo4j to Pi4jv2 and continue to identify required hardware-code modifications.  

In my opinion, it would be nice if this feature stayed inside Pi4j, because Pi4j presents a complete set of features on top of the chosen platform.

We will soon start reviewing Robo4j units functionality, which could give more insights about this topic.

**Comment by FDelporte** [comment]
Closing this discussion/poll. Let's mark it deprecated in V3

**Reply by FDelporte** [reply]
And will be removed in V4.
